src/pages/workspaces/workspace/WorkspaceContainer.js

```diff
@@ -156,6 +156,11 @@ const useCloudEnvironmentPolling = namespace => {
   const [persistentDisks, setPersistentDisks] = useState()
   const [galaxyDataDisks, setGalaxyDataDisks] = useState()
 
+  //clean up timeout on unmount.
+  useEffect(() => {
```

Comment:
the application had a memory leak before we specify cleanup here. I'm not super familiar with useEffect, but it seems that more of effectful timeout logic should be housed in this somehow. Currently, its semantically equivalent to `Utils.useOnUnmount`, but it seems like the effect method of cleanup is preferred. Since in its [description](https://reactjs.org/docs/hooks-effect.html#example-using-classes-1) it discourages the use of onMount/onUnmount

Reply:
Hmm.. this doesn't seem to be quite correct... need to do some further digging.. probably will revert this and do it in a separate PR

Reply:
Ahh... it seems useOnMount is an alias for useEffect.... may not actually be an issue... though I wonder why its still reporting a memory leak from around here...

Reply:
Hmm.. it seems we are passing the function to set up this timeout around to a lot of components, only ensuring we clean up when this component is unmounted... definitely a problem for another PR